chain/state/statetree.go

```diff
@@ -145,22 +196,32 @@ func (st *StateTree) DeleteActor(addr address.Address) error {
 
 	addr = iaddr
 
-	delete(st.actorcache, addr)
-
-	if err := st.root.Delete(context.TODO(), string(addr.Bytes())); err != nil {
-		return xerrors.Errorf("failed to delete actor: %w", err)
+	_, err = st.GetActor(addr)
+	if err != nil {
+		return err
 	}
 
+	st.snaps.deleteActor(addr)
+
 	return nil
 }
 
 func (st *StateTree) Flush(ctx context.Context) (cid.Cid, error) {
```

Comment:
Can save for a separate PR, but - we could be returning the cache from a flush call, tied to the state root. Then when processing a chain we could be reusing the cache instead of having to rebuild it at each epoch

Reply:
I like that, that will pair nicely with the next caching thing, which is caching ID lookups